File: backend/src/main/java/com/ecommerce/app/service/user/UserService.java
package com.ecommerce.app.service.user;

import com.ecommerce.app.dto.user.UserResponse;
import com.ecommerce.app.dto.user.UserUpdateRequest;
import com.ecommerce.app.model.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface UserService {

    UserResponse getUserById(Long id);

    UserResponse updateProfile(Long id, UserUpdateRequest request);

    UserResponse getCurrentUserProfile(Long id);

    Page<UserResponse> getAllUsers(Pageable pageable);

    void deleteUser(Long id);

    User findByEmail(String email);
}

File: frontend/src/services/userService.js
import { api } from './api'

export const userService = {
  getProfile: () => api.get('/users/me'),

  updateProfile: (userData) => api.put('/users/me', userData),

  getAllUsers: (page = 0, size = 10) => api.get(`/users?page=${page}&size=${size}`),

  getUserById: (id) => api.get(`/users/${id}`),

  deleteUser: (id) => api.delete(`/users/${id}`),
}
